src/ui/assets/AssetsUpload.js

```diff
@@ -6,7 +6,9 @@ import { Button } from 'patternfly-react';
 
 import UploadAssetModalContainer from 'ui/assets/modals/upload-assets/UploadAssetModalContainer';
 
-const AssetsUpload = ({ onDrop, buttonVersion }) => {
+const AssetsUpload = ({
+  onDrop, buttonVersion, onAssetSelected, name,
```

Comment:
Having "magic" flags like `buttonVersion` is **highly discouraged**.
This is an explanatory article: https://spicefactory.co/blog/2019/03/26/how-to-avoid-the-boolean-trap-when-designing-react-components/

Correct me if I'm wrong, but what the flag does is choosing which component will trigger the upload flow: it could be the upload button, as well as the dropzone area in the digital assets screen.

If so, in `AssetUpload` my suggestion is removing that flag and adding a property that will contain the component the user will interact with to start the upload flow (in our case they're the upload button and the dropzone component). It will make the component "open for extension, closed for modification" as in https://en.wikipedia.org/wiki/SOLID. 
Let's imagine another another scenario that will trigger the upload flow from another component: with the existing approach, we will need to add a new flag or change the existing one to an enum. 
If we rather pass the "trigger" component as a prop, we won't have anything to change except that prop.